src/libstd/sync/condvar.rs

```diff
@@ -236,6 +263,36 @@ impl StaticCondvar {
         }
     }
 
+    /// Wait on this condition variable for a notification, timing out after a
+    /// specified duration.
+    ///
+    /// The implementation will repeatedly wait while the duration has not
+    /// passed and the function returns `false`.
+    ///
+    /// See `Condvar::wait_timeout_with`.
+    #[unstable]
+    pub fn wait_timeout_with<'a, T, F>(&'static self,
+                                       mut guard: MutexGuard<'a, T>,
+                                       dur: Duration,
+                                       mut f: F)
+                                       -> LockResult<(MutexGuard<'a, T>, bool)>
+            where F: FnMut(&mut MutexGuard<'a, T>) -> bool {
+        // This could be made more efficient by pushing the implementation into sys::condvar
+        let start = SteadyTime::now();
+        while !f(&mut guard) {
+            let now = SteadyTime::now();
+            let consumed = &now - &start;
```

Comment:
This is some interesting information we're kinda just throwing away when returning from this function, although I'm not sure if there's a great way, nor that it's worth it, to return it from this function. (again, just food-for-thought)